pkg/types/vsphere/validation/platform.go

```diff
@@ -221,3 +227,159 @@ func validateLoadBalancer(lbType configv1.PlatformLoadBalancerType) bool {
 		return false
 	}
 }
+
+// validateHosts.
+func validateHosts(p *vsphere.Platform, installConfig *types.InstallConfig, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	// Validate hosts counts match desired replicas
+	allErrs = append(allErrs, validateHostsCount(p.Hosts, installConfig, fldPath)...)
+
+	// Iterate through hosts
+	for _, host := range p.Hosts {
+		// Check Role (bootstrap, control-plane, compute)
+		allErrs = append(allErrs, validateHostRole(host, fldPath)...)
+
+		// Check failure domain (must exist in failure domains)
+		if host.FailureDomain != "" {
+			allErrs = append(allErrs, validateHostFailureDomain(host, p.FailureDomains, fldPath)...)
+		}
+
```

Comment:
Should validateHostsCount() or validateHostRole() check if there is a at least 1 host with role == Bootstrap?

Reply:
For every host.FailureDomain, we are iterating through p.FailureDomains. Would this be more efficient if put p.FailureDomain.Name into a temporary slice and then look for the host's failureDomain using slices.Contains()?